app/scripts/metamask-controller.js

```diff
@@ -1425,6 +1395,7 @@ export default class MetamaskController extends EventEmitter {
       process.env.REJECT_INVALID_SNAPS_PLATFORM_VERSION;
 
     this.snapController = new SnapController({
+      dynamicPermissions: ['endowment:caip25'],
```

Comment:
Lets not forget to follow-up to change the default for this value in the `SnapController`. Could you create an issue for this in the snaps repo, if one doesn't exist already?